[PATCH] From ca66350aecf7db3354019480d11754fabae3a97c Mon Sep 17

From: Stepan Blyschak <stepanb@nvidia.com>
Date: Thu, 20 Oct 2022 13:07:18 +0000
 00:00:00 2001 Subject: [PATCH 09/13] ignore route from default table

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>
---
 zebra/dplane_fpm_nl.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/zebra/dplane_fpm_nl.c b/zebra/dplane_fpm_nl.c
index 41f603a9e..74225e6cb 100644
--- a/zebra/dplane_fpm_nl.c
+++ b/zebra/dplane_fpm_nl.c
@@ -965,6 +965,15 @@ static int fpm_nl_enqueue(struct fpm_nl_ctx *fnc, struct zebra_dplane_ctx *ctx)
 		|| op == DPLANE_OP_NH_UPDATE))
 		return 0;
 
+	/*
+	 * Ignore route from default table, because when mgmt port goes down,
+	 * zebra will remove the default route and causing ASIC to blackhole IO.
+	 */
+	if (dplane_ctx_get_table(ctx) == RT_TABLE_DEFAULT) {
+		zlog_debug("%s: discard default table route", __func__);
+		return 0;
+	}
+
 	nl_buf_len = 0;
 
 	/*
